var/spack/repos/builtin/packages/mapl/package.py

```diff
@@ -254,6 +256,8 @@ class Mapl(CMakePackage):
     depends_on("mpi")
     depends_on("hdf5")
     depends_on("netcdf-c")
+    depends_on("zstd", when="+zstd")
```

Comment:
I think line 259 isn't needed if what you said in the PR description is correct (i.e. netcdf needs zstd, not mapl itself)

Reply:
Agree, the line after ('netcdf-c +zstd') should be sufficient

Reply:
Okay. Update incoming!

I guess next question is: Is there a better way to express this:
```python
    depends_on("netcdf-c")
    depends_on("netcdf-c +zstd", when="+zstd")
```
It just looks...awkward to me.

Reply:
It's actually fairly common to need to identify specs with variants in this manner.  For example, if a package has an `mpi` variant and needs to pass that along to its dependencies.  The same can apply for `cuda` and `rocm` variants.  You can see it in action in packages like `warpx`, `vtk*`, `xsdk*` and others.

The only difference is there isn't typically a space in front of the dependencies' variant:

```
    depends_on("netcdf-c+zstd", when="+zstd")
```

Not sure if that would reduce the awkwardness for you.

Reply:
If it's okay, I'm fine with it.